pylabnet/launchers/logger/log_display.py

```diff
@@ -0,0 +1,210 @@
+    GUI_PORT = 5678
+    GUI_NAME = 'Logger GUI Server'
+
+    def __init__(self, *args, **kwargs):
+
+        self.log_service = None
+        self.log_server = None
+        self.gui_logger = None
+        self.gui_service = None
+        self.gui_server = None
+        self.client_list = {}
+        self.port_list = {}
+        self.script_list = {}
+        self.disconnection = False
+        self.current_output = ''
+
+        sys.stdout = StringIO()
+
+        # Instantiate GUI application
+        self.app = QtWidgets.QApplication(sys.argv)
+        self.main_window = Window(self.app, gui_template=self.LOGGER_UI)
+
+        self._start_logger()
+        self._initialize_gui()
+
+    def start_gui_server(self, gui_port=None):
+
+        # Assign the port to use
+        if gui_port is None:
+            gui_port = self.GUI_PORT
+
+        # Try to connect to the logger
+        try:
+            self.gui_logger = LogClient(
+                host='localhost',
+                port=self.LOG_PORT,
+                module_tag=self.GUI_NAME
+            )
+        except ConnectionRefusedError:
+            raise
+
+        # Instantiate GUI server and update GUI with port details
+        self.gui_service = Service()
+        self.gui_service.assign_module(module=self.main_window)
+        self.gui_service.assign_logger(logger=self.gui_logger)
+        self.gui_server = GenericServer(
+            service=self.gui_service,
+            host='localhost',
+            port=gui_port
+        )
+        self.gui_server.start()
+        self.main_window.gui_label.setText('GUI Port: {}'.format(gui_port))
+
+        # Update internal attributes and add to list of log clients
+        self.client_list[self.GUI_NAME] = QtWidgets.QListWidgetItem(self.GUI_NAME)
+        self.port_list[self.GUI_NAME] = [port for port in self.log_server._server.clients][0]
+        self.main_window.client_list.addItem(self.client_list[self.GUI_NAME])
+        self.client_list[self.GUI_NAME].setToolTip(self.log_service.client_data[self.GUI_NAME])
+
+    def update_terminal(self):
+        """ Updates terminal output on GUI """
+
+        self.main_window.terminal.append(sys.stdout.getvalue())
+        try:
+            self.main_window.terminal.moveCursor(QtGui.QTextCursor.End)
+        except TypeError:
+            pass
+        sys.stdout.truncate(0)
+        sys.stdout.seek(0)
+
+    def check_disconnection(self):
+        """ Checks if a client has disconnected and raises a flag if so"""
+
+        if 'Client disconnected' in sys.stdout.getvalue() or self.disconnection:
+            self.disconnection = True
+
+    def disconnect(self):
+        """ Handles case where client has disconnected """
+
+        to_del = [client for (client, port) in self.port_list.items() if port not in self.log_server._server.clients]
+        for client in to_del:
+            print('[INFO] {} disconnected at {}'.format(client, time.strftime("%Y-%m-%d, %H:%M:%S", time.gmtime())))
```

Comment:
No sure I understand where this print-statement gets printed. The '[INFO]' tag usually get automatically added by outputting to a logger, so it's quite unusual to have it explicitly written out in a print statement.